src/contracts/GPv2Settlement.sol

```diff
@@ -100,24 +106,128 @@ contract GPv2Settlement {
+            encodedTrades.tradeAtIndex(i).decodeTrade(
+                domainSeparator,
+                tokens,
+                trade
+            );
+            processTrade(
+                trade,
+                clearingPrices[trade.sellTokenIndex],
+                clearingPrices[trade.buyTokenIndex],
+                inTransfers[i],
+                outTransfers[i]
+            );
+        }
+    }
+
+    /// @dev Process trades for a single EOA order.
+    /// @param trade The trade to process.
+    /// @param sellPrice The price of the order's sell token.
+    /// @param buyPrice The price of the order's buy token.
+    /// @param inTransfer Memory location to set computed tranfer into the
+    /// settlement contract to execute trade.
+    /// @param outTransfer Memory location to set computed transfer out to order
+    /// owner to execute trade.
+    function processTrade(
+        GPv2Encoding.Trade memory trade,
+        uint256 sellPrice,
+        uint256 buyPrice,
+        GPv2AllowanceManager.Transfer memory inTransfer,
+        GPv2AllowanceManager.Transfer memory outTransfer
+    ) internal pure {
+        GPv2Encoding.Order memory order = trade.order;
+        // NOTE: Currently, the above instanciation allocates an unitialized
+        // `Order` that gets never used. Adjust the free memory pointer to free
+        // the unused memory by subtracting `sizeof(Order) == 288` bytes.
+        // <https://solidity.readthedocs.io/en/v0.7.5/internals/layout_in_memory.html>
+        // TODO: Remove this once the following fix is merged and released:
+        // <https://github.com/ethereum/solidity/pull/10341>
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            mstore(0x40, sub(mload(0x40), 288))
+        }
+
+        inTransfer.owner = trade.owner;
+        inTransfer.token = order.sellToken;
+        outTransfer.owner = trade.owner;
+        outTransfer.token = order.buyToken;
+
+        if (order.kind == GPv2Encoding.OrderKind.Sell) {
+            uint256 executedSellAmount;
+            if (order.partiallyFillable) {
+                executedSellAmount = trade.executedAmount;
+            } else {
+                executedSellAmount = order.sellAmount;
+            }
+
+            inTransfer.amount = executedSellAmount;
+            // NOTE: Don't use `SafeMath.div` here as it allocates a string even
+            // if it does not revert.
+            outTransfer.amount = executedSellAmount.mul(buyPrice) / sellPrice;
+        } else {
+            uint256 executedBuyAmount;
+            if (order.partiallyFillable) {
+                executedBuyAmount = trade.executedAmount;
+            } else {
+                executedBuyAmount = order.buyAmount;
+            }
+
+            // NOTE: Don't use `SafeMath.div` here as it allocates a string even
+            // if it does not revert.
+            inTransfer.amount = executedBuyAmount.mul(sellPrice) / buyPrice;
```

Comment:
`sellPrice` and `buyPrice` are defined by the solver, and I believe so far there is no check where we guarantee limit prices are respected.
I think these checks must happen somewhere in the code introduced in this PR, since 1. after this function call any order info is lost and 2. we don't want to decode the order again. 
Even if you plan to do it in another PR, can you add a todo to the code with this important detail?